File: frontend/src/types/incident.ts
import type { IncidentStatus, Severity, TimelineEventType } from './domain'
import type {
  TriageResult,
  InvestigationResult,
  RootCauseResult,
  RemediationResult,
  HumanDecision,
  PostMortem,
} from './agents'
import type { Alert } from './alert'

export type TimelinePayload =
  | TriageResult
  | InvestigationResult
  | RootCauseResult
  | RemediationResult
  | HumanDecision
  | PostMortem

export interface TimelineEvent {
  id: string
  incident_id: string
  timestamp: string
  stage: IncidentStatus
  type: TimelineEventType
  title: string
  payload: TimelinePayload
}

export interface IncidentSummary {
  id: string
  status: IncidentStatus
  severity: Severity
  service: string
  environment: string
  summary: string
  created_at: string
  updated_at: string
}

export interface Incident extends IncidentSummary {
  alert: Alert
  timeline: TimelineEvent[]
  resolved_at: string | null
}

File: api/app/schemas/incident.py
from typing import Any

from pydantic import BaseModel


class TimelineEvent(BaseModel):
    id: str
    incident_id: str
    timestamp: str
    stage: str
    type: str
    title: str
    payload: dict[str, Any]


class IncidentSummary(BaseModel):
    id: str
    status: str
    severity: str
    service: str
    environment: str
    summary: str
    created_at: str
    updated_at: str


class Incident(IncidentSummary):
    alert: dict[str, Any]
    timeline: list[TimelineEvent]
    resolved_at: str | None
